src/Fixer/Phpdoc/PhpdocAddMissingParamAnnotationFixer.php

```diff
@@ -253,7 +253,8 @@ private function prepareArgumentInformation(Tokens $tokens, int $start, int $end
                     CT::T_CONSTRUCTOR_PROPERTY_PROMOTION_PROTECTED,
                     CT::T_CONSTRUCTOR_PROPERTY_PROMOTION_PUBLIC,
                 ])
-                || (\defined('T_READONLY') && $token->isGivenKind(T_READONLY))
+                || (\defined('T_READONLY') && $token->isGivenKind(T_READONLY)) // @TODO: simplify condition when PHP 8.1+ is required
+                || (\defined('T_PRIVATE_SET') && $token->isGivenKind([T_PRIVATE_SET, T_PROTECTED_SET, T_PUBLIC_SET])) // @TODO: simplify condition when PHP 8.4+ is required
```

Comment:
```suggestion
                || (\PHP_VERSION_ID >= 8_04_00 && $token->isGivenKind([T_PRIVATE_SET, T_PROTECTED_SET, T_PUBLIC_SET]))
```

more readable, no comment needed and more opcache friendly

Reply:
> more readable

That's subjective

> no comment needed

We do `\defined('TOKEN')` _ comment all over the codebase, when it comes to tokens not present in PHP 7.4, so 👎🏼 for the suggested change

> more opcache friendly

how come?

Reply:
> how come?

`PHP_VERSION_ID` is known value constant on compile time. So opcache compiles the expression and evaluates the AND condition to be always false - the opcache then completely then removes the condition from compiled AST.